src/main/scala/akka/persistence/cassandra/query/journal/EventsByPersistenceIdPublisher.scala

```diff
@@ -0,0 +1,201 @@
+package akka.persistence.cassandra.query.journal
+
+import java.nio.ByteBuffer
+import java.lang.{Long => JLong}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+import akka.actor.Props
+import akka.persistence.PersistentRepr
+import akka.persistence.query.EventEnvelope
+import akka.serialization.SerializationExtension
+import com.datastax.driver.core.utils.Bytes
+import com.datastax.driver.core.{ResultSet, Row}
+
+private[journal] object EventsByPersistenceIdPublisher {
+  def props(
+      persistenceId: String,
+      fromSeqNr: Long,
+      toSeqNr: Long,
+      refreshInterval: Option[FiniteDuration],
+      maxBufSize: Long,
+      config: CassandraReadJournalConfig): Props =
+    Props(new EventsByPersistenceIdPublisher(persistenceId, fromSeqNr, toSeqNr, refreshInterval, maxBufSize, config))
+}
+
+// TODO: Decouple database.
+// TODO: Query index tables instead.
+// TODO: Generic message iterator to handle different tables etc.
+private[journal] class EventsByPersistenceIdPublisher(
+    persistenceId: String,
+    fromSeqNr: Long,
+    toSeqNr: Long,
+    refreshInterval: Option[FiniteDuration],
+    maxBufSize: Long,
+    override val config: CassandraReadJournalConfig)
+  extends QueryActorPublisher[EventEnvelope, Long](refreshInterval, maxBufSize)
+  with CassandraReadStatements {
+
+  import config._
+
+  private[this] val cluster = clusterBuilder.build
+  private[this] val session = cluster.connect()
```

Comment:
`Cluster` and `Session` instances are thread-safe and can/should be shared according to the Java driver API docs. Move them to an Akka extension or another shared object?